feather/server/src/systems/player_leave.rs

```diff
@@ -127,14 +127,21 @@ fn create_player_data(
             .to_vec()
             .iter()
             .enumerate()
-            .filter_map(|(slot, maybe_item)| maybe_item.as_ref().map(|item| (slot, item)))
+            .filter(|(_, item)| item.is_filled())
+            .map(|(slot, item)| {
+                if let libcraft_items::InventorySlot::Filled(item) = item {
+                    (slot, item)
+                } else {
+                    unreachable!()
```

Comment:
Please add a comment stating why this is unreachable.

Reply:
Done.

Reply:
Use filter map instead.